test/e2e_env/multizone/localityawarelb/locality_aware_multizone_hybrid_gateway.go

```diff
@@ -0,0 +1,187 @@
+			Setup(multizone.Global)).To(Succeed())
+		Expect(WaitForMesh(mesh, multizone.Zones())).To(Succeed())
+
+		// Kubernetes Zone 1
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(democlient.Install(democlient.WithMesh(mesh), democlient.WithNamespace(namespace))).
+			Install(testserver.Install(
+				testserver.WithName("test-server"),
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-zone-1"),
+			)).
+			Setup(multizone.KubeZone1)).ToNot(HaveOccurred())
+
+		// Universal Zone 5
+		Expect(NewClusterSetup().
+			Install(DemoClientUniversal(
+				"demo-client_locality-aware-lb-gateway_svc",
+				mesh,
+				WithTransparentProxy(true),
+			)).
+			Install(TestServerUniversal("test-server-zone-5", mesh,
+				WithServiceName("test-server_locality-aware-lb-gateway_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-zone-5"}),
+			)).
+			Setup(multizone.UniZone2),
+		).To(Succeed())
+
+		// Universal Zone 4
+		Expect(NewClusterSetup().
+			Install(GatewayProxyUniversal(mesh, "edge-gateway")).
+			Install(TestServerUniversal("gateway-client", mesh, WithoutDataplane())).
+			Install(TestServerUniversal("test-server-zone-4", mesh,
+				WithServiceName("test-server_locality-aware-lb-gateway_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-zone-4"}),
+			)).
+			Setup(multizone.UniZone1),
+		).To(Succeed())
+
+		Expect(multizone.Global.Install(meshGateway)).To(Succeed())
+		Expect(multizone.Global.Install(meshGatewayRoute)).To(Succeed())
+	})
+
+	E2EAfterAll(func() {
+		Expect(multizone.UniZone1.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.UniZone2.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.KubeZone1.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(multizone.Global.DeleteMesh(mesh)).To(Succeed())
+	})
+
+	It("should route based on defined strategy when making requests through gateway", func() {
+		// no lb priorities
+		Eventually(func() (map[string]int, error) {
+			gatewayIP := multizone.UniZone1.GetApp("edge-gateway").GetIP()
+			return client.CollectResponsesByInstance(multizone.UniZone1, "gateway-client", fmt.Sprintf("http://%s", net.JoinHostPort(gatewayIP, "8080")), client.WithHeader("Host", "example.kuma.io"), client.WithNumberOfRequests(100))
+		}, "2m", "10s").Should(
+			And(
+				HaveKeyWithValue(Equal(`test-server-zone-4`), BeNumerically("~", 50, 10)),
+				HaveKeyWithValue(Equal(`test-server-zone-5`), BeNumerically("~", 25, 10)),
+				HaveKeyWithValue(Equal(`test-server-zone-1`), BeNumerically("~", 25, 10)),
+			),
+		)
+
+		// apply lb policy
+		// kuma-4 - priority 0, kuma-5 - priority 1, kuma-1 - priority 2
+		Expect(multizone.Global.Install(YamlUniversal(meshLoadBalancingStrategyDemoClient))).To(Succeed())
+
+		Eventually(func() (map[string]int, error) {
+			gatewayIP := multizone.UniZone1.GetApp("edge-gateway").GetIP()
```

Comment:
nitpick: I guess the IP won't change right?

Reply:
It should not, changed